src/vidpf.rs

```diff
@@ -0,0 +1,735 @@
+    }
+}
+
+impl BitXor for &Proof {
+    type Output = Proof;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        Proof(zip(&self.0, &rhs.0).map(|(a, b)| a ^ b).collect())
+    }
+}
+
+#[derive(Debug)]
+/// PublicKey is used by aggregation servers.
+pub struct PublicKey<R>
+where
+    R: Range,
+    for<'a> &'a R: RangeOps<R>,
+{
+    cw: Vec<CorrectionWord<R>>,
+    cs: Vec<Proof>,
+}
+
+#[derive(Debug)]
+/// Key is the aggreagation server's private key.
+pub struct Key {
+    id: ServerID,
+    value: Vec<u8>,
+}
+
+impl Key {
+    /// generates a key of n bytes at random.
+    pub fn gen(id: ServerID, n: usize) -> Result<Self, Box<dyn Error>> {
+        let mut value = vec![0; n];
+        getrandom::getrandom(&mut value)?;
+        Ok(Key { id, value })
+    }
+}
+
+struct Sequence(Seed, ControlBit, Seed, ControlBit);
+
+impl BitXor for &Sequence {
+    type Output = Sequence;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        Sequence(
+            &self.0 ^ &rhs.0,
+            self.1 ^ rhs.1,
+            &self.2 ^ &rhs.2,
+            self.3 ^ rhs.3,
+        )
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+/// ControlBit
+pub struct ControlBit(Choice);
+
+impl From<u8> for ControlBit {
+    fn from(b: u8) -> Self {
+        ControlBit(Choice::from(b))
+    }
+}
+
+impl From<&Key> for ControlBit {
+    fn from(k: &Key) -> Self {
+        ControlBit::from(k.id as u8)
+    }
+}
+
+impl Fill for ControlBit {
+    fn fill(&mut self, r: &mut Box<dyn RngCore>) {
+        let mut b = [0u8; 1];
+        r.fill_bytes(&mut b);
+        *self = ControlBit::from(b[0] & 0x1)
+    }
+}
+
+impl ConditionallySelectable for ControlBit {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        ControlBit((a.0 & !choice) | (b.0 & choice))
+    }
+}
+
+impl BitAnd<&Seed> for ControlBit {
+    type Output = Seed;
+    fn bitand(self, rhs: &Seed) -> Self::Output {
+        Seed(
+            rhs.0
+                .iter()
+                .map(|x| u8::conditional_select(&0, x, self.0))
+                .collect(),
+        )
+    }
+}
+
+impl BitAnd<&Proof> for ControlBit {
+    type Output = Proof;
+    fn bitand(self, rhs: &Proof) -> Self::Output {
+        Proof(
+            rhs.0
+                .iter()
+                .map(|x| u8::conditional_select(&0, x, self.0))
+                .collect(),
+        )
+    }
+}
+
+impl BitAnd<&Sequence> for ControlBit {
+    type Output = Sequence;
+
+    fn bitand(self, rhs: &Sequence) -> Self::Output {
+        Sequence(self & &rhs.0, self & rhs.1, self & &rhs.2, self & rhs.3)
+    }
+}
+
+impl BitAnd for ControlBit {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self::Output {
+        ControlBit(self.0 & rhs.0)
+    }
+}
+
+impl BitXor for ControlBit {
+    type Output = Self;
+
+    fn bitxor(self, rhs: Self) -> Self::Output {
+        ControlBit(self.0 ^ rhs.0)
+    }
+}
+
+#[derive(Debug, Clone)]
+/// Seed
+pub struct Seed(Vec<u8>);
```

Comment:
We should be able to use `crate::vdaf::xof::Seed` instead.

Reply:
done